x-pack/libbeat/docs/aws-credentials-config.asciidoc

```diff
@@ -1,6 +1,6 @@
 [float]
 === AWS Credentials Configuration
-To configure AWS credentials, there are two different ways supported:
+To configure AWS credentials, there two different ways supported. Look at the examples below, which demonstrate the use of Beats standard AWS authentication parameters using the metricbeat AWS module.
```

Comment:
I think it's odd to show a Metricbeat example in the Heartbeat docs. We should use attributes (variables) or conditional coding to provide examples that are specific to Heartbeat (let me know if you need help with this). 

After changing the example, I'd shorted the lead-up paragraph. It's not clear what's meant by two ways.  I would say something like:

"To configure AWS credentials, either put the credentials into the {beatname_uc} configuration, or use a shared credentials file, as shown in the following examples."

Reply:
@dedemorton you're right, the metricbeat examples look totally weird. The only practical way to make this work would be to conditionally include examples from each implementation. 

This would be a burden for each new AWS credentials using implementation, they'd have to define multiple includes showing their syntax. Additionally, any change to the credentials implementation would also need to edit all those includes. Maybe we should make the shared docs more generic and remove all the metricbeat module stuff?

@kaiyan-sheng WDYT?

Reply:
The complexity depends on how much the examples vary. If you can point me to a few examples, I can help you decide the best approach forward. It's always a tradeoff. Conditional sections make the docs harder to maintain and more fragile, but providing users with relevant info is pretty important. Anyhow, let me know if I can help.